### PR TITLE
Readd Scroll Sync

### LINKED ISSUE #212: reimplement synchronous scrolling
**Describe the solution you'd like**
The editor and preview windows should scroll synchronously again. We need to implement a (new) method to realize that. 

**Describe alternatives you've considered**
Since not re implementing this feature is not really an option, there is no real alternative.

**Additional context**
Currently this is done via many hidden Elements in the rendered DOM which show from which line in the markdown the elements originate from. 

**scrollspy initalization** 
https://github.com/codimd/server/blob/6462968e84e8d92292dd23764a9e558d7800147d/public/js/pretty.js#L85

**scrollspy extras removed**
https://github.com/codimd/server/blob/20a1563d7f81d0454533c569b6505c71a720956f/public/js/extra.js#L600

**smoothHashScroll function**
https://github.com/codimd/server/blob/20a1563d7f81d0454533c569b6505c71a720956f/public/js/extra.js#L781

### PR DESCRIPTION
**Pull request by DerMolly**
### Component/Part
Editor

### Description
This PR adds synchronous scrolling between the codemirror and the preview pane.

### Steps

- [ ] added implementation
- [ ] added / updated tests
- [ ] added / updated documentation
- [ ] extended changelog

### Related Issue(s)
fixes #212

### CODE REVIEW

**Comment by mrdrogdrog** [comment]
@DerMolly please review it, too.

File: src/components/common/splitter/splitter.tsx [hunk 1]
```diff
@@ -54,7 +54,7 @@ export const Splitter: React.FC<SplitterProps> = ({ containerClassName, left, ri
         </div>
       </ShowIf>
       <ShowIf condition={showRight}>
-        <div className='splitter right'>
+        <div className={'splitter right'}>
```
Comment (DerMolly) [inline src/components/common/splitter/splitter.tsx:57]: is there a special reason for these unnecessary brackets?

File: src/components/editor/markdown-renderer/markdown-it-plugins/line-number-marker.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      const lineNumber = tokens[index].attrGet('data-linenumber')
+      if (!lineNumber) {
+        // don't render broken linemarkers without a linenumber
+        return ''
+      }
+      // noinspection CheckTagEmptyBody
+      return `<codimd-linemarker data-linenumber='${lineNumber}'></codimd-linemarker>`
+    }
+  }
+}
```
Comment (DerMolly) [inline src/components/editor/markdown-renderer/markdown-it-plugins/line-number-marker.ts:55]: this file should be cleaned up. one could use earlier returns and a avoid wrapping everything in an if

**Comment by mrdrogdrog** [comment]
Closed in favor of #386

**Pull request closed without merging**